pkg/apis/v1/duration.go

```diff
@@ -28,6 +32,16 @@ const Never = "Never"
 // that the duration is disabled and sets the inner duration as nil
 type NillableDuration struct {
 	*time.Duration
+
+	// Raw is used to ensure we remarshal the NillableDuration in the same format it was specified.
+	// This ensures tools like Flux and ArgoCD don't mistakenly detect drift due to our conversion webhooks.
+	Raw []byte `hash:"ignore"`
+}
+
+func MustParseNillableDuration(val string) NillableDuration {
+	nd := NillableDuration{}
+	lo.Must0(json.Unmarshal([]byte(fmt.Sprintf("%q", val)), &nd))
```

Comment:
```suggestion
        // Use %q instead of %s to ensure that we unmarshal the value as a string and not an int
	lo.Must0(json.Unmarshal([]byte(fmt.Sprintf("%q", val)), &nd))
```